src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceDataRegistryKey.cs

```diff
@@ -74,23 +82,42 @@ public byte[] GetValue(string name, bool usePool)
                 }
                 sizeInput = size;
 
-                ReleaseBlob(blob, usePool);
-                blob = CreateBlob(size, usePool);
+                ReleaseData(data, usePool);
+                data = CreateBlob(size, usePool);
             }
 
             if (ret != 0)
             {
+                ReleaseData(data, usePool);
                 Win32Error(ret, name);
             }
 
-            return blob;
+            return data;
         }
 
-        public void ReleaseBlob(byte[] blob, bool usePool = true)
+        public void ReleaseData(byte[] data, bool usePool = true)
         {
             if (usePool)
             {
-                ArrayPool<byte>.Shared.Return(blob);
+#if DEBUG
+                // Fill array by garbage to detect early situations when this array is used after return
+                MemoryMarshal.Cast<byte, uint>(data.AsSpan()).Fill(0xBAADF00D);
+#endif
+
+                ArrayPool<byte>.Shared.Return(data);
+            }
+        }
+
+        public void Close()
+        {
+            Dispose();
+        }
+
+        public void Dispose()
+        {
+            if (!_hkey.IsInvalid)
```

Comment:
@jkotas it seems (from safehandle.cpp) that Dispose() only does anything if the handle is not invalid, and it sets the handle to invalid. Is it safe to rely on that, and eliminate the IsInvalid check here? It doesn't seem to be documented.

Reply:
Yes. The contract of all Dispose methods is that it is safe to call them multiple times.

Reply:
Ah, of course that's a general contract pointed out on IDisposable docs.